### PR TITLE
Add info on how to extend webpack.config.js

### PR DESCRIPTION
**Pull request by mor10**
See #14560 for background

## Description

Adds explanation and code example for how to extend `webpack.config.js` using `spread` operator.

## Types of changes

Documentation update.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

**Comment by gziolo** [comment]
On a different note, I was looking into https://github.com/smooth-code/svgr and I wasn't aware it has so many useful packages in addition to CLI tool and webpack loader. It makes me wonder if we shouldn't redefine the way we handle SVG icons and maybe start using files with SVGs and tackle them with custom Babel loader which would convert them to React components on the fly providing a unified format with both React web and React Native can process.

**Comment by gziolo** [comment]
There is related chat in core-js WordPress Slack channel (link requires registration at https://make.wordpress.org/chat/):
https://wordpress.slack.com/archives/C5UNMSU4R/p1553509272322700

I think it's fine to include this note, we might need to rephrase it to ensure that people are aware that doing so they can run into compatibility issues when the default config changes because of WordPress needs or when webpack changes API.

File: packages/scripts/README.md [hunk 1]
```diff
@@ -287,4 +287,28 @@ Should there be any situation where you want to provide your own webpack config,
...
+      ...defaultConfig.module.rules,
+      {
+        test: /\.svg$/,
+        use: ["@svgr/webpack", "url-loader"]
+      }
+    ]
+  }
+};
+```
+
```
Comment (nosolosw) [inline packages/scripts/README.md]: A similar version of this was in the original PR, but we decided against it as to not promote this practice. See also relevant [core-js](https://make.wordpress.org/core/2019/03/13/javascript-chat-summary-march-12th-2019/) discussions. It is already proved convenient for others, though, so we may want to revisit that decission and document this while we figure out a better way to extend it.

I'd like us to add a notice along these lines:

```suggestion
If you follow this approach, please, be aware that future versions of this package may change what webpack and Babel plugins we bundle, default configs, etc. Should those changes be necessary, they will be registered in the [package's CHANGELOG](https://github.com/WordPress/gutenberg/blob/master/packages/scripts/CHANGELOG.md), so make sure to read it before upgrading.
```

**Review by mkaz — APPROVED** [review]
With @nosolosw  suggestion this looks good to go

**Comment by mor10** [comment]
@gziolo I would back the inclusion of SVGR 100%. It makes adding and using SVGs in blocks significantly easier and removes the need for custom workarounds. Here's how I'm using it to add a block logo as a React component:

```javascript
import customLogoURL, { ReactComponent as customLogo } from "./logo.svg";

registerBlockType("block/myblock", {
  title: __("A Block", "myblocs"),
  icon: {
    src: customLogo // The React component
  },
  category: "regular",
  attributes: {
    blockImage: {
      type: "string",
      default: customLogoURL // The URL to the original SVG
    }
  },
  (...)
}
```

**Comment by gziolo** [comment]
> I would back the inclusion of SVGR 100%. It makes adding and using SVGs in blocks significantly easier and removes the need for custom workarounds.

Thanks for sharing. I opened a follow-up issue to keep what you shared documented (#14628).

**Comment by isvictorious** [comment]
Any examples of adding support for SCSS?

**Comment by gziolo** [comment]
> Any examples of adding support for SCSS?

In #13814 I listed a few webpack configs from WordPress plugins:

* [Gutenberg SlotFill and Filter demos](https://github.com/ryanwelcher/slotfill-and-filter-demos/blob/master/webpack.config.js) by @ryanwelcher 
* [Juxtapose Block](https://github.com/mkaz/juxtapose-block/blob/master/webpack.config.js) by @mkaz 
* [Gutenberg Custom Fields](https://github.com/youknowriad/gcf/blob/master/webpack.config.js) by @youknowriad 
* [Gutenberg Extensibility Workshop](https://github.com/youknowriad/gutenberg-extensibility-workshop/blob/master/webpack.config.js) by @youknowriad 
* [Drop it](https://github.com/youknowriad/dropit/blob/master/webpack.config.js) by @youknowriad 
* [Tomodomo Block Starter](https://github.com/TomodomoCo/wp-block-starter/blob/master/webpack.config.js) by @chrisvanpatten 

You might find there an example of how to handle SCSS. We don't use webpack to handle SCSS files in Gutenberg. However, this might be a good idea to include such capability in `wp-scripts start` and `wp-scripts build`.

I think I also saw SASS integration in [create-cloud-block](https://github.com/front/create-cloud-block) (A boilerplate generator for building Gutenberg Cloud blocks to use with Drupal and WordPress):
https://github.com/front/create-cloud-block/blob/master/examples/1-simple-block/webpack.config.js

**Comment by fabiankaegy** [comment]
@isvictorious I played around with it some and this is what I am currently using to extend the config to use scss compiling

https://github.com/fabiankaegy/callout-block/blob/extending-wp-scripts-scss/webpack.config.js

**Comment by gziolo** [comment]
@fabiankaegy - it looks reasonable as far as I can tell. In Gutenberg there is also this file:
https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/theme.scss

I don't remember what role it plays, but it might be worth to double check if you would need it as well.

**Comment by isvictorious** [comment]
@gziolo @fabiankaegy thanks for sharing. As we build more and more blocks I am trying to avoid dependency management as much as possible. @wordpress/scripts has most of what we need (testing, babel, etc) except Sass. 

A big issue going forward with custom blocks is going to be dependency management.

**Comment by mor10** [comment]
Counterpoint: Sass is falling out of favor fast in the wider web community for good reason: between custom properties and nesting as native CSS features, and PostCSS doing everything Sass can do and more, Sass is seen to stand in the way of progress. I would recommend setting up a PostCSS config instead.

**Comment by gziolo** [comment]
@fabiankaegy, thanks for opening an issue, let's continue the discussion there 👍
